lib/vizkit/vizkit_items.rb

```diff
@@ -803,6 +803,22 @@ def write(&block)
         end
     end
 
+    class TaskModelItem < VizkitItem
+        attr_reader :model
+
+        def initialize(task, options = Hash.new)
+            super()
+            @options = Kernel.validate_options options,:item_type => :label
+            @model = task.to_async.model
```

Comment:
You can't do that here, there's no guarantee that you can resolve the task (it failed when I tried it).

You need to enclose the model access through an `on_reachable` block

```
diff --git a/lib/vizkit/vizkit_items.rb b/lib/vizkit/vizkit_items.rb
index 121070e..8a98425 100644
--- a/lib/vizkit/vizkit_items.rb
+++ b/lib/vizkit/vizkit_items.rb
@@ -809,10 +809,14 @@ module Vizkit
         def initialize(task, options = Hash.new)
             super()
             @options = Kernel.validate_options options,:item_type => :label
-            @model = task.to_async.model
-
             if @options[:item_type] == :value
-                setText @model.name
+                task.on_reachable do
+                    begin
+                        @model = task.to_async.model
+                        setText @model.name
+                    rescue
+                    end
+                end
             elsif @options[:item_type] == :label
                 setText "TaskModel"
             end
```